src/main/java/com/denizenscript/depenizen/bukkit/commands/mythicmobs/MythicSkillCommand.java

```diff
@@ -3,113 +3,107 @@
-        }
-        scriptEntry.defaultObject("power", new ElementTag(0));
+    public void addCustomTabCompletions(TabCompletionsBuilder tab) {
+        tab.add((Set<String>) MythicMobsBridge.getSkillNames());
     }
 
-    @Override
-    public void execute(ScriptEntry scriptEntry) {
-        List<EntityTag> casters = (List<EntityTag>) scriptEntry.getObject("casters");
-        ElementTag skill = scriptEntry.getObjectTag("skill");
-        List<EntityTag> entity_targets = (List<EntityTag>) scriptEntry.getObject("entity_targets");
-        List<LocationTag> location_targets = (List<LocationTag>) scriptEntry.getObject("location_targets");
-        ElementTag power = scriptEntry.getObjectTag("power");
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), db("casters", casters), skill, db("location_Targets", location_targets), db("entity_targets", entity_targets), power);
-        }
-        ArrayList<Entity> entityTargets = null;
-        ArrayList<Location> locationTargets = null;
-        if (entity_targets != null) {
-            entityTargets = new ArrayList<>();
-            for (EntityTag entity : entity_targets) {
-                entityTargets.add(entity.getBukkitEntity());
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgLinear @ArgName("skill") ElementTag skill,
+                                   @ArgLinear @ArgName("targets") ListTag targets,
+                                   @ArgPrefixed @ArgName("power") @ArgDefaultText("1") ElementTag power,
+                                   @ArgPrefixed @ArgName("casters") @ArgDefaultNull ListTag casters,
+                                   @ArgPrefixed @ArgName("trigger") @ArgDefaultNull EntityTag trigger,
+                                   @ArgPrefixed @ArgName("origin") @ArgDefaultNull LocationTag origin,
+                                   @ArgPrefixed @ArgName("parameters") @ArgDefaultNull MapTag parameters,
+                                   @ArgPrefixed @ArgName("variables") @ArgDefaultNull MapTag variables,
+                                   @ArgLinear @ArgName("async") @ArgDefaultText("false") boolean async) {
+        List<Entity> entityTargets = new ArrayList<>();
+        List<Location> locationTargets = new ArrayList<>();
+        for (ObjectTag object : targets.objectForms) {
+            if (object.canBeType(EntityTag.class)) {
+                entityTargets.add(((EntityTag) object).getBukkitEntity());
+            }
+            else if (object.canBeType(LocationTag.class)) {
+                locationTargets.add((Location) object);
             }
         }
-        else {
-            locationTargets = new ArrayList<>(location_targets);
+        if (!entityTargets.isEmpty() && !locationTargets.isEmpty()) {
+            Debug.echoError("Cannot have both entity and location targets.");
```

Comment:
`Debug#echoError(ScriptEntry,String)`

Reply:
That's only needed in async commands, which this is not